pkg/api/models/keptn_context_extended_c_e.go

```diff
@@ -55,3 +56,20 @@ func (ce *KeptnContextExtendedCE) DataAs(out interface{}) error {
 	}
 	return json.Unmarshal(bytes, out)
 }
+
+// Validate checks whether the required properties 'time', 'type', 'id' and 'source' are defined and non-empty
+func (ce *KeptnContextExtendedCE) Validate() error {
+	if ce.Time.IsZero() {
```

Comment:
is `Time` always defined?

Reply:
yes, since it is not specified as a pointer in the `KeptnContextExtendedCE` struct, it will always have an initialized value - when it was not present in an incoming event, it will thus have the zero-value